From c6900f5dbe001497cecd6aeed5062f8d8bc34617 Mon Sep 17 00:00:00 2001
From: Prerna Saxena <prerna@linux.vnet.ibm.com>
Date: Sun, 15 Feb 2015 09:48:00 +0530
Subject: [PATCH 1/3] PowerPC : Make 'qemu-system-ppc64' the default emulator
 on ppc64[le].

PowerPC : Explicitly associate 'qemu-system-ppc64' as the
 default emulator for all 64-bit PowerPC guests ( both Big & Little Endian )

Signed-off-by: Prerna Saxena <prerna@linux.vnet.ibm.com>
(cherry picked from commit bdbe723fcd8b55678b0ad984134f7b79d6fbeb89)
Signed-off-by: Chris J Arges <chris.j.arges@canonical.com>
---
 src/qemu/qemu_capabilities.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/qemu/qemu_capabilities.c b/src/qemu/qemu_capabilities.c
index 13f3cd3..c427ca2 100644
--- a/src/qemu/qemu_capabilities.c
+++ b/src/qemu/qemu_capabilities.c
@@ -672,9 +672,14 @@ virQEMUCapsFindBinaryForArch(virArch hostarch,
                              virArch guestarch)
 {
     char *ret;
-    const char *archstr = virQEMUCapsArchToString(guestarch);
+    const char *archstr;
     char *binary;
 
+    if (ARCH_IS_PPC64(guestarch))
+        archstr = virQEMUCapsArchToString(VIR_ARCH_PPC64);
+    else
+        archstr = virQEMUCapsArchToString(guestarch);
+
     if (virAsprintf(&binary, "qemu-system-%s", archstr) < 0)
         return NULL;
 
-- 
1.9.1

